hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java

```diff
@@ -93,13 +93,28 @@ UserDetailsService userDetailsService() {
         if (userPrincipals.isEmpty()) {
             final String name = securityProperties.getUser().getName();
             final String password = securityProperties.getUser().getPassword();
-            userPrincipals.add(new UserPrincipal(name, password, name, name, name, null, DEFAULT_TENANT,
-                    PermissionUtils.createAllAuthorityList()));
+            List<GrantedAuthority> authorityList;
+            if (securityProperties.getUser().getRoles().isEmpty()) {
```

Comment:
Better extract `securityProperties.getUser().getRoles()` to a local variable and put it next to the username and password variable. Than your if-statement looks cleaner:

```java
if(roles.isEmpty()){
   authorityList = PermissionUtils.createAllAuthorityList();
} else {
   authorityList = createAuthoritiesFrom(roles);
} 
```